docs/resources/instance.md

```diff
@@ -141,6 +141,25 @@ The `file` block supports:
 * `create_directories` - *Optional* - Whether to create the directories leading
 	to the target if they do not exist.
 
+The `exec` block supports:
+
+* `command` - **Required** - The command to be executed (list of strings).
+
+* `triggers` - *Optional* - A list of arbitrary strings that, when changed, will force the command
+  to be rerun.
```

Comment:
To recap our yesterday's conversation. The tricky part about special conditions is that we need to evaluate the plan of an instance before the update without doing any modification on the infrastructure itself. This means that any checks that need to be executed within an instance, would require instance to be running, which we *can not* ensure during the planning phase. For VMs, even simple file require the instance to be running.

Technically, we could ensure the instance is running during the planning phase by simply starting it if it is stopped, but this goes against Terraform concepts because we are doing an *update* without prior approval from the user. In addition, planning phase could take up to a minute instead of a few seconds.

Another approach would be to treat those special conditions as *requires update*, which means that such instances would always show *plan changes*, even if the conditions are already satisfied. Conditions would then be evaluated only during an actual update.

However, running a command on each apply can be already achieved using triggers:
```hcl
exec {
  command  = ["..."]
  triggers = [timestamp()] # This will ensure the change is always detected.
}
```

With such approach we can add new conditions that will determine whether command needs to be run, but they should not trigger any plan changes, because we cannot verify whether the change is already satisfied.

For advance use cases, the easiest approach would be to upload Puppet/Ansible configuratio to an instance (using `file` block) and then apply it using `exec` block that has a `timestamp()` trigger.